main.go

```diff
@@ -1,62 +1,284 @@
+			Env:      "PUPPET_CERT",
+			Argument: "cert",
+			Usage:    "path to the SSL certificate PEM file signed by your site's Puppet CA",
+			Value:    &handler.puppetCert,
+		},
+		&sensu.PluginConfigOption{
+			Path:     "key",
+			Env:      "PUPPET_KEY",
+			Argument: "key",
+			Usage:    "path to the private key PEM file for that certificate",
+			Value:    &handler.puppetKey,
+		},
+		&sensu.PluginConfigOption{
+			Path:     "cacert",
+			Env:      "PUPPET_CACERT",
+			Argument: "cacert",
+			Usage:    "path to the site's Puppet CA certificate PEM file",
+			Value:    &handler.puppetCACert,
+		},
+		&sensu.PluginConfigOption{
+			Path:     "insecure-skip-tls-verify",
+			Env:      "PUPPET_INSECURE_SKIP_TLS_VERIFY",
+			Argument: "insecure-skip-tls-verify",
+			Usage:    "skip SSL verification",
+			Value:    &handler.puppetInsecureSkipVerify,
+		},
+		{
+			Path:      "sensu-api-url",
+			Env:       "SENSU_API_URL",
+			Argument:  "sensu-api-url",
+			Shorthand: "u",
+			Default:   "http://localhost:8080",
+			Usage:     "The Sensu API URL",
+			Value:     &handler.sensuAPIURL,
+		},
+		{
+			Path:      "sensu-api-key",
+			Env:       "SENSU_API_KEY",
+			Argument:  "sensu-api-key",
+			Shorthand: "a",
+			Usage:     "The Sensu API key",
+			Value:     &handler.sensuAPIKey,
+		},
+		{
+			Path:      "sensu-ca-cert",
+			Env:       "SENSU_CA_CERT",
+			Argument:  "sensu-ca-cert",
+			Shorthand: "c",
+			Usage:     "The Sensu Go CA Certificate",
+			Value:     &handler.sensuCACert,
 		},
-	}
-
-	options = []*sensu.PluginConfigOption{
-		&handlerConfigOptions.Example,
 	}
 )
 
 func main() {
-	handler := sensu.NewGoHandler(&handlerConfig.PluginConfig, options, checkArgs, executeHandler)
+	handler := sensu.NewGoHandler(&handler.PluginConfig, options, validate, executeHandler)
 	handler.Execute()
 }
 
-func checkArgs(_ *types.Event) error {
-	if len(handlerConfig.Example) == 0 {
-		return fmt.Errorf("--example or HANDLER_EXAMPLE environment variable is required")
+func validate(event *types.Event) error {
+	// Make sure we have a valid event
+	if event.Check == nil || event.Entity == nil {
+		return errors.New("invalid event")
+	}
+
+	// Make sure all required options are provided
+	if len(handler.endpoint) == 0 {
+		return errors.New("the PuppetDB API endpoint is required")
 	}
+	if len(handler.puppetCert) == 0 {
+		return errors.New("the path to the SSL certificate is required")
+	}
+	if len(handler.puppetKey) == 0 {
+		return errors.New("the path to the private key is required")
+	}
+	if len(handler.sensuAPIURL) == 0 {
+		return errors.New("the Sensu API URL is required")
+	}
+	if len(handler.sensuAPIKey) == 0 {
+		return errors.New("the Sensu API key is required")
+	}
+
+	// Make sure the PuppetDB endpoint URL is valid
+	u, err := url.Parse(handler.endpoint)
+	if err != nil {
+		return fmt.Errorf("invalid PuppetDB API endpoint URL: %s", err)
+	}
+	if u.Scheme == "" || u.Host == "" {
+		return errors.New("invalid PuppetDB API endpoint URL")
```

Comment:
Can you not assume `https` for the scheme?

I agree it's best to give the endpoint as a valid `scheme://host:port/path` URI, but I think a lot of people might just pass in `host/path` and get that `invalid PuppetDB API endpoint URL` error without it being crystal clear that they actually need to pass in `https://host/path`.

What do you think?